client/state/support/actions.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Internal dependencies
+ */
+import {
+	SUPPORT_USER_ACTIVATED,
+	SUPPORT_USER_DEACTIVATED,
+	SUPPORT_USER_FETCH_TOKEN,
+} from 'state/action-types';
+
+import User from 'lib/user';
+
+import userSettings from 'lib/user-settings';
+
+/**
+ * Requests a support user token, then dispatches the relevant actions upon response
+ * 
+ * @param  {string} supportUser     Support username
+ * @param  {string} supportPassword Support password
+ * @return {thunk}                  The action thunk
+ */
+export function supportUserFetchToken( supportUser, supportPassword ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: SUPPORT_USER_FETCH_TOKEN,
+			supportUser
+		} );
+
+		if ( supportUser && supportPassword ) {
+			let user = new User();
+
+			user.changeUser(
+				supportUser,
+				supportPassword,
+				( error ) => {
+					if ( error ) {
+						dispatch( supportUserDeactivated( error.message ) );
+					} else {
+						let userData = Object.assign( {}, user.data );
+						dispatch( supportUserActivated( userData ) );
+					}
```

Comment:
we're five levels deep here - maybe we could break these into functions and pass the args by themselves

```js
const user = new User();
const deactivateSupportUser = ( { message } ) => dispatch( supportUserDeactivated( message );
const activateSupportUser = userData => dispatch( supportUserActivated( userData ) );
const changeUser = error => error
	? deactivateSupportUser( error )
	: activateSupportUser( Object.assign( {}, user.data ) );

user.changeUser( supportUser, supportPassword, changeUser, deactivateSupportUser );
```

or again using some point-free syntax

```js
const User = new User();
const getErrorMessage = ( { message } ) => message;
const deactivateSupportUser = compose( dispatch, supportUserDeactivated, getMessage );
const activateSupportUser = compose( dispatch, supportUserActivated, clone );
const conditional = ( c, truthy, falsey ) => !! c ? truthy : falsey;
const changeUser = partialRight( conditional, deactivateSupportUser, activateSupportUser );

user.changeUser( supportUser, supportPassword, changeUser, deactivateSupportUser );
```

Of course, I recommend the first approach to the second for use here in Calypso, but thought I'd throw the second out there anyway